### ISSUE TITLE
Consider alternative syntax

### ISSUE THREAD

**Issue by ericnewton76** [commandlineparser/commandline#165]
<a href="https://github.com/Korporal"><img src="https://avatars0.githubusercontent.com/u/12262952?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [Korporal](https://github.com/Korporal)**
_Monday Sep 04, 2017 at 15:28 GMT_
_Originally opened as https://github.com/gsscoder/commandline/issues/476_

----

Hi,

Consider this generic approach, I think it's better for developers, simpler pattern:

```
static void Main(string[] args) {
  
    var options = CommandLine.Parser.Default.ParseArguments<Options>(args);

    if (options.Available)
       // Values are available here
       if (options.Verbose) Console.WriteLine("Filename: {0}", options.InputFile);
 
}
```

just a thought...

**Comment by johnjaylward** [comment]
Fixed in #634. Example:
```c#
var options = CommandLine.Parser.Default.ParseArguments<Options>(args).Value;
if (options != default) {
    // ... do something with your options
}
```

If you want both the parsed options and the errors, you would do something like this:
```c#
var parsed = CommandLine.Parser.Default.ParseArguments<Options>(args);
if (parsed.Value != default) {
    // ... do something with your options
} else {
   // ... do something with Parsed.Errors
}
```

**Comment by johnjaylward** [comment]
@Korporal, not sure if you still care, but since you opened this issue initially, I thought I'd tag you that a PR has been created that is similar to your requested syntax

**Comment by moh-hassan** [comment]
merged #634 into develop
